File: js/my_practice/classes/playerClass.js
class Player {
  constructor(name, type) {
    console.log("Player", this);
    this.name = name;
    this.type = type;
  }

  introduce() {
    console.log(`Hi I am ${this.name}, I am a ${this.type}`);
  }
}

class Wizard extends Player {
  constructor(name, type) {
    super(name, type);
    console.log("Wizard", this);
  }
  play() {
    console.log(`Oyaaa, I'm ${this.type}`);
  }
}

const wizard1 = new Wizard("James", "Healer");
const wizard2 = new Wizard("John", "Doe");

console.log(wizard1.play());
console.log(wizard2.introduce());

File: python/my_practice/classes/playerClass.py
from abc import ABC

# Parent Class
class Player:
    def __init__(self, name, type) -> None:
        self.name = name
        self.type = type
        
    def introduce(self):
        return f"Hi, I'm {self.name}, {self.type}"
       
# Sub Class
class Wizard(Player):
    def __init__(self, name, type) -> None:
        super().__init__(name, type)
        print("Wizard", self)
        
    def play(self):
        return f"Oyaa, I'm {self.type}"
        
wizard1 = Wizard("James", "Healer")
wizard2 = Wizard("John", "Doe")